state/aws/dynamodb/dynamodb.go

```diff
@@ -393,6 +393,61 @@ func (d *StateStore) parseTTL(req *state.SetRequest) (*int64, error) {
 	return nil, nil
 }
 
+// Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail.
+func (d *StateStore) Multi(ctx context.Context, request *state.TransactionalStateRequest) error {
+	twinput := &dynamodb.TransactWriteItemsInput{
+		TransactItems: []*dynamodb.TransactWriteItem{},
+	}
+
+	// Dedup ops where the last operation with a matching Key takes precedence
+	txs := map[string]int{}
+	for i, o := range request.Operations {
+		txs[o.GetKey()] = i
+	}
+
+	for i, o := range request.Operations {
+		// skip operations removed in simulated set
+		if txs[o.GetKey()] != i {
+			continue
+		}
+
+		twi := &dynamodb.TransactWriteItem{}
+		switch req := o.(type) {
+		case state.SetRequest:
+			value, err := d.marshalToString(req.Value)
+			if err != nil {
+				return fmt.Errorf("dynamodb error: failed to marshal value for key %s: %s", req.Key, err)
+			}
+			twi.Put = &dynamodb.Put{
```

Comment:
Do we also need to take care of Update op?

Reply:
When I looked at the conformance tests, I only saw `SetRequest` and `DeleteRequest` being tested and looks like is the same pattern across the State components that support Multiple Tx Ops.

Reply:
I don't believe we have an Update op

Reply:
Ok, got it.